src/OptimistAttestationResolver.sol

```diff
@@ -76,54 +73,47 @@ contract OptimistAttestationResolver is
     /// @inheritdoc SchemaResolverUpgradeable
     function onAttest(
         Attestation calldata attestationInput,
-        uint256
+        uint256 value
     ) internal
     whenNotPaused
     override(SchemaResolverUpgradeable, AllowlistResolverUpgradeable)
     returns (bool)
     {
-        require(!allowedAttesters[attestationInput.attester], "OptimistAttestationResolver: attester is not allowed");
-        require(attestationUidByRecipient[attestationInput.recipient] == bytes32(0), "OptimistAttestationResolver: recipient already record by uid");
+        require(AllowlistResolverUpgradeable.onAttest(attestation, value), "OptimistAttestationResolver: attester is not allowed");
+        require(attestationUidByRecipient[attestationInput.recipient] == bytes32(0), "OptimistAttestationResolver: recipient already has an attestation");
         attestationUidByRecipient[attestationInput.recipient] = attestationInput.uid;
-        optimist.mint(attestationInput.recipient);
```

Comment:
Coupling the OptimistNFT logic in here is not ideal because this implies that this contract (OptimistAttestationResolver) needs to know the implementation details of OptimistAllowlist and OptimistNFT.

Ideally, the flow should be a one way read

OptimistNFT == (reads) ==> OptimistAllowlist == (reads) ==> OptimistAttestationResolver. with the lower level contracts unaware of the existence of the higher level contracts